From 3000bdb8d18fdd8be747be96ecfbf0512613c0d9 Mon Sep 17 00:00:00 2001
From: Yu Jiahua <yujiahua1@huawei.com>
Date: Thu, 20 Jan 2022 12:22:36 +0800
Subject: [PATCH] Revert "sched: Fix branch prediction error in static_key"
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 3000bdb8d18fdd8be747be96ecfbf0512613c0d9


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4QU5Z?from=project-issue
CVE: NA

--------------------------------

This patch revert ias feature from open-euler kernel.

This reverts commit 37870264c163bc05b2d04f0c07021f6381b24846.

Signed-off-by: Yu Jiahua <Yujiahua1@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 kernel/sched/fair.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 96e8b479c88c..6ad5d878223e 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -8576,7 +8576,7 @@ static void update_blocked_averages(int cpu)
 	update_rq_clock(rq);
 
 #ifdef CONFIG_SCHED_OPTIMIZE_LOAD_TRACKING
-	if (!static_branch_likely(&sched_blocked_averages)) {
+	if (!static_branch_unlikely(&sched_blocked_averages)) {
 		rq_unlock_irqrestore(rq, &rf);
 		return;
 	}
-- 
2.26.2

